niworkflows/utils/bids.py

```diff
@@ -209,6 +227,10 @@ def collect_data(
     if echo:
         queries["bold"]["echo"] = echo
 
+    if session_id is not False:
+        for acq in queries.keys():
+            queries[acq]["session"] = session_id
```

Comment:
Since `bids_filters` also updating the default entities, I'm not sure which of these should have the final say..

Reply:
If we put `session` directly in the `layout.get` call like with `subject`, then we can just let `**query` raise an exception if `session` is multiply defined.